models/provenance.py

```diff
@@ -6,6 +6,8 @@
 from sqlalchemy.orm import relationship
 from sqlalchemy.dialects.postgresql import JSONB
 
+from pipeline.utils import get_git_hash
```

Comment:
This is causing an error in the tests ; looks like a circular import error.  We should fix that before merging ths.

Reply:
I was afraid I would have circular imports. Let me take care of that first.